src/posts/introducing-pint-xarray/index.md

```diff
@@ -0,0 +1,292 @@
+
+## Quantifying with pint-xarray
+
+The easiest way to create a unit-aware xarray object is to use the helper package we made: [pint-xarray](https://github.com/xarray-contrib/pint-xarray).
+Once you `import pint_xarray` you can access unit-related functionality via `.pint` on any xarray DataArray or Dataset (this works via [xarray's accessor interface](https://xarray.pydata.org/en/stable/internals/extending-xarray.html)).
+
+Above we have seen examples of quantifying explicitly, where we specify the units in the call to `.quantify()`.
+We can do this for multiple variables too, and we can also pass `pint.Unit` instances:
+
+```python
+ds = xr.Dataset({'a': 2, 'b': 10})
+
+ds.pint.quantify({'a': 'kg',
+                  'b': pint.Unit('moles')})
+```
+
+```
+Out:
+<xarray.Dataset>
+Dimensions:  ()
+Data variables:
+    a        int64 [kg] 2
+    b        int64 [mol] 10
+```
+
+Alternatively, we can quantify from the object's `.attrs`, automatically reading the metadata which xarray objects carry around.
+If nothing is passed to `.quantify()`, it will attempt to parse the `.attrs['units']` entry for each data variable.
+
+This means that for scientific datasets which are stored as files with units in their attributes (which netCDF and Zarr can do for example), using pint with xarray becomes as simple as
+
+```python
+import pint_xarray
+
+ds = open_dataset(filepath).pint.quantify()
+```
+
+## Dequantifying
+
+To convert our pint arrays back into numpy arrays, we can use `.dequantify`.
+This will strip the units from the arrays and replace them into the `.attrs['units']` of each variable.
+This is useful when we want to save our data back to a file, as it means that the current units will be preserved in the attributes of a netcdf file (or zarr store etc.), as long as we just do `ds.pint.dequantify().to_netcdf()`.
+
+## Dask integration
+
+So xarray can wrap dask arrays, and now it can wrap pint quantities… Can we use both together? Yes!
+
+You can get a unit-aware, dask-backed array either by `.pint.quantify()`-ing a chunked array, or you can `.pint.chunk()` a quantified array.
+(If you have dask installed, then `open_dataset(f).pint.quantify()` will already give you a dask-backed, quantified array.)
+From there you can `.compute()` the dask-backed objects as normal, and the units will be retained.
+
+(Under the hood we now have an `xarray.DataArray` wrapping a `pint.Quantity`, which wraps a `dask.array.Array`, which wraps a `numpy.ndarray`.
+This "multi-nested duck array" approach can be generalised to include other array libraries (e.g. `scipy.sparse`), but requires [co-ordination](https://github.com/pydata/duck-array-discussion) between the maintainers of the libraries involved.)
+
+## Unit-aware indexes
+
+We would love to be able to promote xarray indexes to pint Quantities, as that would allow you to select data subsets in a unit-aware manner like
```

Comment:
```suggestion
We would love to be able to promote Xarray indexes to Pint Quantities, as that would allow you to select data subsets in a unit-aware manner like:
```